zaza/openstack/charm_tests/vault/utils.py

```diff
@@ -280,6 +281,65 @@ def ensure_secret_backend(client):
         pass
 
 
+def wait_until_ready(clients):
+    """Wait until vault is ready (for all clients).
+
+    Raise an error if it's not ready within a reasonable time.
+
+    :param clients: Clients to use to talk to vault
+    :type clients: List[CharmVaultClient]
+    :raises: AssertionError
+    """
+    for client in clients:
+        for attempt in tenacity.Retrying(
+            reraise=True,
+            wait=tenacity.wait_fixed(10),
+            stop=tenacity.stop_after_attempt(12),  # wait for max 2 minutes
```

Comment:
So using `10` and `12` as bare literals *once* is probably okay, but using them multiple times through the code starts to look a bit suspect.  I'd be inclined to add some consts to either the class or the module for them if they are repeated, but there are no hard-and-fast rules about this, so it's also okay to not do that (as they aren't really configurable).